### DISCUSSION TITLE
Is it possible to use iModel.js without iModelHub?

### DISCUSSION THREAD

**Discussion by KonstantinKliukach** [iTwin/itwinjs-core#125] [Q&A] [answered]
Hello!

We are looking for some tool for showing BIM's and some other models.

So, I have some questions:
1. Is it possible to use imodel.js without imodelhub? Cause, we can't share our models, fortunately.
2. Or may be you have some variants for commercial usage?

**Comment by jan-slegr** [comment]
Hi Konstantin,

> Is it possible to use imodel.js without imodelhub?

Yes, you can create [static iModel snapshot](https://www.imodeljs.org/learning/backend/accessingimodels/). But without iModelHub, some features are not accessible.

**Comment by kabentley** [comment, marked as answer]
Think of iModelHub as "GitHub for iModels". The features of iModel.js that have to do with creating/pushing/pulling changesets require iModelHub, obviously (it's where you push/pull them to/from).

But, as Jan points out, you can use all of the visualization (either Desktop via Electron or web-based) and snapshot creation tools without a project in iModelHub. If you download the free  [iTwin Synchronizer](https://www.bentley.com/en/products/product-line/digital-twins/itwin-synchronizer), there's a way to create a local snapshot without iModelHub. We're working now on a desktop viewer to be included in the iTwin Synchronizer download.

There is also a commercial model, analogous to GitHub Enterprise, if you want to create a private iModelHub. 

HTH,
Keith

**Reply by KonstantinKliukach** [reply]
Thank you for reply!

**Comment by Josh-Schifter** [comment]
The easiest way to create snapshots is using the [iTwin Snapshot](https://autoupdatecdn.bentley.com/itsnp/client/iTwinSnapshot.exe) application.  It works 100% offline, no sign-in required.  Once you've created a snapshot, it guides you to a free desktop viewer.

Thanks,
Josh
